src/borg/archiver/prune_cmd.py

```diff
@@ -26,26 +26,33 @@
 logger = create_logger()
 
 
-def unique_func():
+# The *_period_func group of functions create period grouping keys to group together archives falling within a certain
+# period. Among archives in each of these groups, only the latest (by creation timestamp) is kept.
+
+
+def unique_period_func():
     counter = 0
 
-    def inner(a):
+    def inner(_a):
+        """Group archives by an incrementing counter, practically making each archive a group of 1"""
         nonlocal counter
         counter += 1
         return counter
 
     return inner
 
 
-def default_period_func(pattern):
+def pattern_period_func(pattern):
     def inner(a):
+        """Group archives by extracting given strftime-pattern from their creation timestamp"""
         # compute in local timezone
         return a.ts.astimezone().strftime(pattern)
 
     return inner
 
 
 def quarterly_13weekly_period_func(a):
+    """Group archives by extracting from their creation timestamp their ISO-8601 week numbers"""
```

Comment:
It's not the week number that is ultimately extracted, but is the "quarter" measured in groups of 13 weeks.